src/compiler/crystal/codegen/primitives.cr

```diff
@@ -366,8 +366,8 @@ class Crystal::CodeGenVisitor
   end
 
   private def codegen_raise_overflow
-    location = @call_location.not_nil!
-    set_current_debug_location(location) if @debug.line_numbers?
+    location = @call_location
```

Comment:
Using `#not_nil!` is intentional here. Having debug location information is mandatory for when debug information is wanted to be emitted. Otherwise the program will fail with `--debug` but run ok, so it was to catch that missing information even when not using `--debug`.

Reply:
That makes sense. However, with this PR this code can now be triggered when there's no call at all. For example:

```crystal
a = 1
f = ->a.+(Int32) # this ends up generation a function that implements + and can raise overflow
f.call(Int32::MAX)
```

The `->a.+(Int32)` can happen multiple times in code so there's no single "call" to use, unlike when you write `a + 1`. I don't know how to handle this...

Reply:
Can't we reach the location of the proc literal when inlining the primitive?

Reply:
We could. But that will not always work correctly because two proc literals that point to the same method will result in the same function being generated, so the location will point to the first literal, which is not always correct.

Maybe we need a reproducible test case to check for when this breaks.

Reply:
Oh, I see. Well the story needs to continue with more support for debugging information then.

Reply:
I'll see what I can do :-)

Reply:
I mean in a future PR, but if you want to tackle that now I wont interfere :-P